activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -89,34 +89,51 @@ def self.quoted_table_names # :nodoc:
         @quoted_table_names ||= {}
       end
 
-      def initialize(connection, logger = nil, config = {}) # :nodoc:
+      def initialize(config_or_deprecated_connection, deprecated_logger = nil, deprecated_connection_options = nil, deprecated_config = nil) # :nodoc:
         super()
 
-        @raw_connection      = connection
-        @owner               = nil
-        @instrumenter        = ActiveSupport::Notifications.instrumenter
-        @logger              = logger
-        @config              = config
-        @pool                = ActiveRecord::ConnectionAdapters::NullPool.new
-        @idle_since          = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+        if config_or_deprecated_connection.is_a?(Hash)
+          @raw_connection = nil
+          @config = config_or_deprecated_connection.symbolize_keys
+          @logger = ActiveRecord::Base.logger
```

Comment:
Are we saying adapters can't set their own logger now? Do any adapters set their own logger? I think that'd be odd but not impossible.

Reply:
My plan is to ultimately pull the connection's base model from the handler (or pass it in the supplied config hash?), and then use that object's logger -- so you can't directly override the adapter's logger during initialize (which hardly anyone calls anyway -- note that `mysql2_connection` and friends are already just passing `logger`), but there is some control beyond the hard-coded AR::Base.